react/features/polls/components/PollDialog.js

```diff
@@ -0,0 +1,406 @@
+        const { dispatch } = this.props;
+
+        dispatch(endPoll());
+    }
+
+    _onChoiceTextSubmit: (Object) => void;
+
+    /**
+     * Function to handle when enter key is pressed in the choice text input
+     * field to automatically add a new choice to the list.
+     *
+     * @param {Object} event - Key press event.
+     * @returns {void}
+     */
+    _onChoiceTextSubmit(event: Object) {
+        if (event.keyCode === 13
+            && event.shiftKey === false) {
+            event.preventDefault();
+
+            this._addChoice();
+        }
+    }
+
+    _onQuestionTextChange: (Object) => void;
+
+    /**
+     * Update the question text in local state.
+     *
+     * @param {event} event - Keyboard event.
+     * @returns {void}
+     */
+    _onQuestionTextChange(event: Object) {
+        const text: string = event.target.value;
+
+        this.setState({
+            question: {
+                ...this.state.question,
+                text
+            }
+        });
+    }
+
+    _onSubmit: () => void;
+
+    /**
+     * Submit button handler.
+     *
+     * @returns {void}
+     */
+    _onSubmit() {
+        const { isPollRunning } = this.props;
+
+        if (isPollRunning) {
+            this._endPoll();
+        } else {
+            this._createNewPoll();
+        }
+    }
+
+    _removeChoice: (number) => void;
+
+    /**
+     * Removes an item from list.
+     *
+     * @param {string} id - ID of item in list.
+     * @returns {void}
+     */
+    _removeChoice(id: string) {
+        const { choices } = this.state;
+
+        // if this is the last available choice, don't remove it.
+        // there must be atleast 1 choice so user can add more
+        if (Object.keys(choices).length === 1) {
+            return;
+        }
+
+        const filteredChoices = { ...choices };
+
+        delete filteredChoices[id];
+
+        this.setState({
+            choices: filteredChoices
+        });
+    }
+
+    /**
+     * Component render method.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { isPollRunning, userID } = this.props;
+        const title = isPollRunning ? 'dialog.endPoll' : 'dialog.startPoll';
+        const tabs = isPollRunning ? [ {
+            component: VoteForm,
+            label: 'polls.vote',
+            props: {
+                userID,
+                onVote: this._vote
+            },
+            styles: 'voteFormContainer',
+            submit: null
+        } ] : [ {
+            component: PollCreateForm,
+            label: 'polls.create',
+            props: {
+                choices: this.state.choices,
+                onKeyDown: this._onChoiceTextSubmit,
+                onChoiceTextChange: this._choiceTextChange,
+                onRemoveChoice: this._removeChoice,
+                onQuestionTextChange: this._onQuestionTextChange
+            },
+            propsUpdateFunction: (oldProps, newProps) => {
+                return { ...newProps };
```

Comment:
Yeah, this is also needed so that the same dialog component use the new props to update its state since the dialog initialize its state from the first props. The child tab receives the list items "poll choices" as props from the dialog -parent.